sdk/lib/rtl/heap.h

```diff
@@ -217,7 +217,8 @@ typedef struct _HEAP_LIST_LOOKUP
     ULONG NumberOfUnCommittedRanges;        \
     USHORT SegmentAllocatorBackTraceIndex;  \
     USHORT Reserved;                        \
-    LIST_ENTRY UCRSegmentList
+    LIST_ENTRY UCRSegmentList;              \
+    PVOID LastEntryInSegment //FIXME: non-Vista
```

Comment:
Yeah it gets removed. I think the Vista+ system has the UCR descriptor itself be right in front of the uncommitted range, therefore the last entry doesn't need to be found... something like that.
However while our data structures are mostly Vista-based, our code is mostly 2003-based and it seemed like it would be a gigantic refactor to do this the Vista way.